index.js

```diff
@@ -0,0 +1,156 @@
+const exec = require('@actions/exec');
+const core = require('@actions/core');
+const github = require('@actions/github');
+const { parse: gitDiffParser } = require('what-the-diff');
+const fs = require('fs-extra');
+const simpleGit = require('simple-git');
+const env = process.env;
+
+function matchExact(r, str) {
+  let match = str.match(r);
+  return match && str === match[0];
+}
+
+function fixNewLineEOF(b) {
+  // Replace all trailing whitespaces
+  b = b.replace(/[ \t\n]*$/, '\n');
+
+  if (b.length === 1) {
+    // if the remaining character is a whitespace
+    if (/[ \t\n]*$/.test(b)) {
+      return '';
+    } else {
+      return b + '\n';
+    }
+  }
+  return b;
+}
+
+async function getChangedFilesPaths(pull_request, octokit, owner, repo) {
+  const { data: pullRequestDiff } = await octokit.pulls.get({
+    owner,
+    repo,
+    pull_number: pull_request.number,
+    mediaType: {
+      format: 'diff'
+    }
+  });
+
+  const parsedDiff = gitDiffParser(pullRequestDiff);
+
+  const changedFilePaths = parsedDiff.map((e) => {
+    return e['newPath'].replace('b/', '');
+  });
+
+  return changedFilePaths;
+}
+
+async function run() {
+  const token = core.getInput('GH_TOKEN');
+  let ignorePaths = core.getInput('IGNORE_FILE_PATTERNS');
+  if (!ignorePaths) {
+    ignorePaths = [];
+  } else {
+    ignorePaths = JSON.parse(ignorePaths);
+  }
+
+  core.info('Ignore File Patterns: ' + JSON.stringify(ignorePaths));
+
+  let commitMessage = core.getInput('COMMIT_MESSAGE');
```

Comment:
can you read the inputs first at the beginning of the function, It will be easy to trace code in future and need not look into entire code to see where all the inputs are read.
```suggestion
  const token = core.getInput('GH_TOKEN');
  let ignorePaths = core.getInput('IGNORE_FILE_PATTERNS');
  let commitMessage = core.getInput('COMMIT_MESSAGE');
  
  
  if (!ignorePaths) {
    ignorePaths = [];
  } else {
    ignorePaths = JSON.parse(ignorePaths);
  }
  core.info('Ignore File Patterns: ' + JSON.stringify(ignorePaths));
  
```